### DISCUSSION TITLE
sync-wave annotation on objects generated by helm chart recognised by argocd for ordering?

### DISCUSSION THREAD

**Discussion by mdfaizsiddiqui** [argoproj/argo-cd#14891] [General]
Hi

I have two helm charts that needs to be deployed in sequential order, meaning all the objects generated by helm chart A should be created first and then chart B should start creating the objects. 

So, to achieve this I have added sync wave annotation to the application objects for both the charts, but that does not have the intended behaviour that was expecting. Both the applications are synched one by one very quickly but the generated objects from both the charts start synching at the same time.

My question is if I add the sync-wave annotation objects to all the objects generated by helm chart A, say 0 and then 1 to all the objects of helm chart B, will that fix the problem? I want to know if argocd recognised annotations on the object generated by helm chart for ordering?

Thanks

**Comment by FernandoMiguel** [comment]
are those two charts, two different apps?
if so, you want to use progressive sync, as sync-wave is only for charts within the same app
https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Progressive-Syncs/

**Reply by mdfaizsiddiqui** [reply]
Two different argo app, each deploying different helm chart (A & B)

**Reply by FernandoMiguel** [reply]
@mdfaizsiddiqui did progressive sync help?

**Comment by vitaly-dt** [comment]
Is there a way to use progressive sync between 2 different application sets?

**Reply by FernandoMiguel** [reply]
AFAIK progressive sync is per applicationset

**Comment by jannfis** [comment]
> My question is if I add the sync-wave annotation objects to all the objects generated by helm chart A, say 0 and then 1 to all the objects of helm chart B, will that fix the problem? I want to know if argocd recognised annotations on the object generated by helm chart for ordering?

Sync waves are only considered in the same sync context, so I'm afraid what you want to achieve here is not quite possible.

There are two potential workarounds:

* Use an app-of-apps pattern, having a parent app that manages the two Applications for the Helm charts. When you sync the parent app, the sync waves on the Applications for the Helm charts will be respected - but only in the sync context of the parent application.
* As @FernandoMiguel pointed out, Progressive Sync with ApplicationSet could be another potential way of solving this.

Both would not prevent the standalone sync of any of the apps, though. We're working on "true" application dependencies, but there's still some way to go for that: https://github.com/argoproj/argo-cd/pull/15280
